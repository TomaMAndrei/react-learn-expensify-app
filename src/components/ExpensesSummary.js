import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from "../selectors/expenses";
import getTotalExpenses from "../selectors/expenses-total";
import numeral from 'numeral';

export class ExpensesSummay extends React.Component{
  expenseCount = () => this.props.expenses.length;
  expensesTotal = () => numeral(getTotalExpenses(this.props.expenses)/ 100).format('$0,0.00');
  render(){
    if(this.expenseCount()>0){
      return (
        <div>
          {`Viewing ${this.expenseCount()} ${this.expenseCount()>1?"expenses":"expense"} totalling ${this.expensesTotal()}`}
        </div>
      );
    }else{
      return null;
    }
    
  }
}
const mapStateToProps = (state) =>({ 
  expenses:selectExpenses(state.expenses,state.filters)
 });

export default connect(mapStateToProps)(ExpensesSummay)