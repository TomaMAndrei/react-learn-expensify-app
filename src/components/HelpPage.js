import React from 'react';

const HelpPage = () => (
  <div>
    This is form help page component
  </div>
);

export default HelpPage;