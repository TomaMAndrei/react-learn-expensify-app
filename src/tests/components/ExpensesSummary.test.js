import React from "react";
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummay } from '../../components/ExpensesSummary';



test("should expenses summary render corectly with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummay expenses={[expenses[1]]} />
  );
  expect(wrapper).toMatchSnapshot();
});
test("should expenses summary render corectly with 3 expenses", () => {
  const wrapper = shallow(
    <ExpensesSummay expenses={expenses} />
  );
  expect(wrapper).toMatchSnapshot();
});

