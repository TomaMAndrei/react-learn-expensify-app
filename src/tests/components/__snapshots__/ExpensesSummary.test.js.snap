// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should expenses summary render corectly with 1 expense 1`] = `
<div>
  Viewing 1 expense totalling $1,095.00
</div>
`;

exports[`should expenses summary render corectly with 3 expenses 1`] = `
<div>
  Viewing 3 expenses totalling $1,141.95
</div>
`;
